src/init.py

```diff
@@ -0,0 +1,126 @@
+from os import getenv
+import shutil
+import subprocess
+from pathlib import Path
+from itertools import product
+import build
+from apscheduler.schedulers.blocking import BlockingScheduler
+from apscheduler.triggers.cron import CronTrigger
+import logging
+import sys
+
+
+def getvar(var: str) -> str:
+    val = getenv(var)
+    if val == "" or val is None:
+        raise ValueError('Environment variable "%s" has an invalid value.' % var)
+    return val
+
+
+def make_key(key_path: str, key_subj: str) -> None:
+    subprocess.run(
+        ["/root/make_key", key_path, key_subj],
+        stdout=subprocess.DEVNULL,
+        stderr=subprocess.DEVNULL,
+        check=True,
+        input="\n".encode(),
+    )
+
+
+def init() -> None:
+    # Copy the user scripts
+    root_scripts = "/root/user_scripts"
+    user_scripts = getvar("USERSCRIPTS_DIR")
+    shutil.copytree(user_scripts, root_scripts)
+
+    # Delete non-root files
+    to_delete = []
+    for path in Path(root_scripts).rglob("*"):
+        if path.isdir(path):
+            continue
+
+        # Check if not owned by root
+        if path.owner != "root":
+            logging.warning("File not owned by root. Removing %s", path)
+            to_delete.append(path)
+            continue
+
+        # Check if non-root can write (group or other)
+        perm = oct(path.stat().st_mode)
+        group_write = perm[-2] > "4"
+        other_write = perm[-1] > "4"
+        if group_write or other_write:
+            logging.warning("File writable by non root users. Removing %s", path)
+            to_delete.append(path)
+
+    for f in to_delete:
+        f.unlink()
+
+    # Initialize CCache if it will be used
+    use_ccache = getvar("USE_CCACHE") == "1"
```

Comment:
should probably also accept "true" here

Reply:
[54b50b8](https://github.com/lineageos4microg/docker-lineage-cicd/pull/342/commits/54b50b8240c158bf66dd7b67f5d8eb969479328f)